lib/irb/cmd/show_cmds.rb

```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 require "stringio"
+require "rdoc/ri/driver"
```

Comment:
Do we need to wrap this with `begin rescue` like irb is doing in completion?
Or this file is delayed loaded so we don't need to rescue it? (maybe the loading timing will change in the future?)

Reply:
The pager implementation part of `rdic/ri/driver` is about 30 lines of 1500 lines.
Maybe we could implement it in IRB? then we can customize the command line option like
```ruby
# https://github.com/ruby/rdoc/blob/5db4767dda4b4320eeac84aba152ab56ec20b839/lib/rdoc/ri/driver.rb#L1470
pagers = [ENV['RI_PAGER'], ENV['PAGER'], 'pager', 'less', 'more']
↓ 
# IRB's ansi format pager
pagers = [ENV['RI_PAGER'], ENV['PAGER'], 'pager --foobar', 'less --raw-control-chars', 'more --raw-control-chars']
```

Reply:
> Do we need to wrap this with begin rescue like irb is doing in completion?

Since we've declared `rdoc` as a dependency in #648, `rdoc` should always be installed along with `irb`. If that didn't happen for some reason, it's something else we need to fix.

Put it in another way: I think we can remove the rescue block in completion and other places.

@hsbt can you help me confirm that if my assumption above is correct? Or when `irb` is used as a default gem, `rdoc` could still be missing? ([example](https://github.com/rails/rails/issues/44143#issuecomment-1013663083))

> Maybe we could implement it in IRB? then we can customize the command line option like

Generally I agree with this direction, but I found a few concerns when implementing it:

- Should `IRB::Pager` also accept `ENV['RI_PAGER']`?
    - If it doesn't, users would get inconsistent behaviour where IRB's pager doesn't pickup their config for RI pager
    - If it does, it sounds like a weird coupling with `rdoc`?
- Should `IRB::Pager` set different pager options (`-r` for example) than `rdoc`'s?
    - This will bring similar concerns the above question

I think if we want to avoid using `rdoc`'s pager, we should do that for all features so we can get rid off of the above issues. For example, IRB should only use `rdoc` to get documentation content and handle the display itself. But at the same time the effort would grow significantly.

Reply:
Because it's necessary to pass `-r` to `less` in order to correctly display IRB's ANSI escape sequences, I eventually decided to implement `IRB::Pager` with the support of `ENV['RI_PAGER']` for now.

I think we can later replace the use of RDoc pager in `show_doc` and doc dialog with something like this:

```rb
name = "String"
driver = RDoc::RI::Driver.new({})
found, klasses, includes, extends = driver.classes_and_includes_and_extends_for(name)
doc = driver.class_document(name, found, klasses, includes, extends)
formatter = RDoc::Markup::ToAnsi.new
contents = doc.accept(formatter)
IRB::Pager.page do |io|
  io.puts contents
end
IRB::Pager.page do |io|
  io.puts contents
end
```